feat(auth): fields validation

koliy82 opened the pull request:
closed #3

koliy82 committed:
feat(auth): fields validation

ripls56 commented on internal/grpc/auth/server.go:
```diff
@@ -87,7 +94,16 @@ func (s *Server) SignUp(
 	}
 
-	err := s.auth.Registration(
+	if !regexp.MustCompile(emailRegex).MatchString(req.Email) {
```
Убери `MustCompile`, он киданет панику если не сможет сбилдить regex.

ripls56 replied:
Замени MustCompile, он киданет панику если не сможет сбилдить regex.

ripls56 commented on internal/grpc/auth/server.go:
```diff
@@ -101,3 +117,45 @@ func (s *Server) SignUp(
 	return &emptypb.Empty{}, nil
 }
+
+func PasswordComplexity(password string) error {
```
Переименовать в `IsPwdComplex` и изменить сигнатуру метода, на func(string) bool.
Если пароль сложный вернуть true, иначе false. Также вынести в pkg `validation`.
Покрыть тестами по [TDT](https://go.dev/wiki/TableDrivenTests)

ripls56 commented on internal/grpc/auth/server.go:
```diff
@@ -71,6 +74,10 @@ func (s *Server) Login(
 }
 
+const (
+	emailRegex = `^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$`
```
Вынести в pkg validation

ripls56 commented:
@koliy82 на счет тестов, там пример в pkg `jwt` есть как писать, тест кейсы TDT'шные может gpt сгенерить, он это неплохо делает

koliy82 committed:
refactor(auth)

Убрал MustCompile
Переименовал PasswordComplexity в IsPwdComplex
Вынес EmailRegex и IsPwdComplex в pgs/validation
TDT сам сделаешь, мне и себе лень тесты писать

ripls56 reviewed (commented):
Осталось тестики

ripls56 committed:
feat(validation): add test for `IsPwdComplex`

ripls56 committed:
refactor(validation): change returned expr

ripls56 committed:
refactor(validation): extract func from `IsPwdComplex`

Extract calculation value of `symbol pool` in another function
